lib/datadog/profiling/profiler.rb

```diff
@@ -49,6 +53,9 @@ def component_failed(failed_component)
           'See previous log messages for details.'
         )
 
+        # We explicitly not stop the crash tracker in this situation, under the assumption that, if a component failed,
+        # we're operating in a degraded state and crash tracking may still be helpful.
```

Comment:
Related to previous minor comment about conflicting models/lifecycles between profiler and crash tracker.

After this, in theory, there could be a reconfiguration of `Datadog` through `Datadog.configure` and thus a restarting of profiler right? But if this were to happen, there's now a dangling crash tracking start that was never stopped and thus all future reconfigurations will fail.

In practice this does sound like enough of a edge case for us to be able to just go with unsupported usage so don't think it's a blocker in any way.

Reply:
> After this, in theory, there could be a reconfiguration of Datadog through Datadog.configure and thus a restarting of profiler right? But if this were to happen, there's now a dangling crash tracking start that was never stopped and thus all future reconfigurations of crash tracker may fail?

Actually, not! This comment is about stopping the profiler via `component_failed`, whereas the components stop the profiler via `shutdown!`, which has been modified to stop the crashtracker.

Thus, the semantics are: if the profiler stops itself because it ran into an issue, the crash tracker states. But once ddtrace decides to clean up the profiler and/or restart it, the crash tracker is correctly updated.

Reply:
I think the current behavior from the native side is quite reasonable/useful: last-writer-wins-singleton means that even if there's a weird bug on our Ruby code the crash tracker can always be started and stopped by a new instance.